express-crud/server.ts

```diff
@@ -0,0 +1,135 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const { grades } = req.body;
+    if (!grades) throw new ClientError(400, 'please provide grades');
+    const sql = `
+select *
+from "grades"
+where "grades" = $1
+`;
+    const result = await db.query(sql, [grades]);
+    const gradesRow = result.rows;
+    res.status(200).json(gradesRow);
+    if (!gradesRow === undefined) {
```

Comment:
There's no need to check this, because `gradesRow` will always be an array. If there are no rows found, it will be an empty array. So you can just return `gradesRow` without checking it.